### PR TITLE
add additional support for custom validation on forms

### PR DESCRIPTION
**Pull request by markhayden**
### Adds custom validation to pathfora forms
Adds the ability to pass a custom Regex pattern alongside all custom form elements. In addition, adds a new class for "us-postal-code" that enforces 5 numeric digits. Example implementation before for validation and testing:

```
<script>
  var customForm = new pathfora.Form({
    id: 'custom-form-5',
    msg: 'custom form',
    layout: 'slideout',
    formElements: [
      {
        type: 'text',
        placeholder: 'Only "z" and "b" Allowed (Exactly 6 Characters)',
        name: 'custom_input',
        pattern: '^[zb]{6}$',
      },
      {
        type: 'us-postal-code',
        placeholder: 'Only 5 Digits Allowed',
        name: 'postal_code',
      },
    ],
  });

  pathfora.initializeWidgets([customForm]);
</script>
```

gif

### CODE REVIEW

**Comment by smcmahon42** [comment]
@markhayden @ashyablok is this something we can pickup in August? Clorox has been relentless with asking for updates so they can expand on their use of the Pathfora JS overrides with experiences. 

CC @kbyler

**Comment by ashyablok** [comment]
@markhayden I'm fine picking this up and completing the PR? unless you have time.

File: dist/pathfora.js [hunk 1]
```diff
@@ -1,270 +1,15 @@
...
 
-    obj.form = {
```
Comment (ashyablok) [inline dist/pathfora.js:74]: I think you need to run this with code coverage off.

**Review by ashyablok — COMMENTED** [review]
Can we add docs for the pattern?

**Review by prosdev — APPROVED** [review]

**Review by ashyablok — APPROVED** [review]

**Pull request merged**